source/fe/mapping_fe.cc

```diff
@@ -909,12 +909,82 @@ MappingFE<dim, spacedim>::transform_real_to_unit_cell(
   const typename Triangulation<dim, spacedim>::cell_iterator &cell,
   const Point<spacedim> &                                     p) const
 {
-  Assert(false, StandardExceptions::ExcNotImplemented());
+  const std::vector<Point<spacedim>> support_points =
+    this->compute_mapping_support_points(cell);
+
+  Tensor<1, spacedim> DF[dim];
+  // [TODO]
+  // When 2nd derivatives are implemented, we'll uncomment this
+  // Tensor<1, spacedim> D2F[dim][dim];
+
+  Point<dim>      p_unit;
+  Point<spacedim> mapped_point;
+  Point<spacedim> p_minus_F;
+
+  Point<dim>     f;
+  Tensor<2, dim> df;
```

Comment:
As far as I can see, of these five variables, only `f` and `p_unit` live outside the loop. Please just move the declarations of the others to the place where they are assigned in the loop.